From 20a6c567bee6b38d71604a852aa68dd178071d9a Mon Sep 17 00:00:00 2001
From: Zubair Lutfullah Kakakhel <Zubair.Kakakhel@imgtec.com>
Date: Fri, 21 Oct 2016 09:42:48 +0100
Subject: [PATCH] dts: ci40: Update dts from openwrt repository

The pistachio dts is fluid at the moment and is maintained in the openwrt
repo as that is what the Ci40 ships with.

Sync with that.

Main issues: Wifi firmware driver and dts entries are slightly out of sync

https://github.com/CreatorDev/openwrt/blob/creator/target/linux/pistachio/dts/pistachio.dtsi
---
 arch/mips/boot/dts/pistachio/pistachio.dtsi | 66 +++++++++++++++++++++--------
 1 file changed, 48 insertions(+), 18 deletions(-)

diff --git a/arch/mips/boot/dts/pistachio/pistachio.dtsi b/arch/mips/boot/dts/pistachio/pistachio.dtsi
index 1c20583..44e43b1 100644
--- a/arch/mips/boot/dts/pistachio/pistachio.dtsi
+++ b/arch/mips/boot/dts/pistachio/pistachio.dtsi
@@ -205,7 +205,6 @@
 		#size-cells = <0>;
 	};
 
-
 	i2c2: i2c@18100400 {
 		compatible = "img,scb-i2c";
 		reg = <0x18100400 0x200>;
@@ -244,16 +243,22 @@
 		compatible = "img,i2s-in";
 		reg = <0x18100800 0x200>;
 		interrupts = <GIC_SHARED 7 IRQ_TYPE_LEVEL_HIGH>;
-		dmas = <&mdc 30 0xffffffff 0>;
-		dma-names = "rx";
 		clocks = <&cr_periph SYS_CLK_I2S_IN>;
 		clock-names = "sys";
 		img,i2s-channels = <6>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&i2s_in_pins>;
+		img,cr-periph = <&cr_periph>;
+		dmas =  <&mdc 30 0xffffffff 0>,
+			<&mdc 24 0xffffffff 0>,
+			<&mdc 25 0xffffffff 0>,
+			<&mdc 26 0xffffffff 0>,
+			<&mdc 27 0xffffffff 0>,
+			<&mdc 28 0xffffffff 0>,
+			<&mdc 29 0xffffffff 0>;
+		dma-names = "rx", "rx0", "rx1", "rx2",
+			    "rx3", "rx4", "rx5";
+		#sound-dai-cells = <1>;
+		#sound-platform-cells = <1>;
 		status = "disabled";
-
-		#sound-dai-cells = <0>;
 	};
 
 	i2s_out: i2s-out@18100a00 {
@@ -462,6 +467,7 @@
 
 			gpio-controller;
 			#gpio-cells = <2>;
+			gpio-ranges = <&pinctrl 0 0 16>;
 
 			interrupt-controller;
 			#interrupt-cells = <2>;
@@ -472,6 +478,7 @@
 
 			gpio-controller;
 			#gpio-cells = <2>;
+			gpio-ranges = <&pinctrl 0 16 16>;
 
 			interrupt-controller;
 			#interrupt-cells = <2>;
@@ -482,6 +489,7 @@
 
 			gpio-controller;
 			#gpio-cells = <2>;
+			gpio-ranges = <&pinctrl 0 32 16>;
 
 			interrupt-controller;
 			#interrupt-cells = <2>;
@@ -492,6 +500,7 @@
 
 			gpio-controller;
 			#gpio-cells = <2>;
+			gpio-ranges = <&pinctrl 0 48 16>;
 
 			interrupt-controller;
 			#interrupt-cells = <2>;
@@ -502,6 +511,7 @@
 
 			gpio-controller;
 			#gpio-cells = <2>;
+			gpio-ranges = <&pinctrl 0 64 16>;
 
 			interrupt-controller;
 			#interrupt-cells = <2>;
@@ -512,6 +522,7 @@
 
 			gpio-controller;
 			#gpio-cells = <2>;
+			gpio-ranges = <&pinctrl 0 80 10>;
 
 			interrupt-controller;
 			#interrupt-cells = <2>;
@@ -616,6 +627,7 @@
 			spim1 {
 				pins = "mfio3", "mfio4", "mfio5";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -623,6 +635,7 @@
 			spim1-quad {
 				pins = "mfio6", "mfio7";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -630,6 +643,7 @@
 			spim1-cs0 {
 				pins = "mfio0";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -637,6 +651,7 @@
 			spim1-cs1 {
 				pins = "mfio1";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -644,6 +659,7 @@
 			spim1-cs1 {
 				pins = "mfio58";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -651,6 +667,7 @@
 			spim1-cs2 {
 				pins = "mfio2";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -658,6 +675,7 @@
 			spim1-cs2 {
 				pins = "mfio31";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -665,6 +683,7 @@
 			spim1-cs2 {
 				pins = "mfio55";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -672,6 +691,7 @@
 			spim1-cs3 {
 				pins = "mfio56";
 				function = "spim1";
+				drive-strength = <2>;
 			};
 		};
 
@@ -679,6 +699,14 @@
 			spim1-cs4 {
 				pins = "mfio57";
 				function = "spim1";
+				drive-strength = <2>;
+			};
+		};
+
+		i2s_in_alt_clk_pins: i2s-in-alt-clk-pins {
+			i2s-in-alt-clk {
+				pins = "mfio11", "mfio12";
+				function = "spis";
 			};
 		};
 
@@ -994,11 +1022,16 @@
 	};
 
 	cr_periph: clk@18148000 {
-		compatible = "img,pistachio-cr-periph", "syscon";
+		compatible = "img,pistachio-cr-periph", "syscon", "simple-bus";
 		reg = <0x18148000 0x1000>;
 		clocks = <&clk_periph PERIPH_CLK_SYS>;
 		clock-names = "sys";
 		#clock-cells = <1>;
+
+		pistachio_reset: reset-controller {
+			compatible = "img,pistachio-reset";
+			#reset-cells = <1>;
+		};
 	};
 
 	cr_top: clk@18149000 {
@@ -1007,12 +1040,6 @@
 		#clock-cells = <1>;
 	};
 
-	pistachio_reset: reset-controller {
-		compatible = "img,pistachio-reset";
-		img,cr-periph = <&cr_periph>;
-		#reset-cells = <1>;
-	};
-
 	fuse@18149200 {
 		compatible = "img,pistachio-efuse";
 		reg = <0x18149200 0x200>;
@@ -1067,11 +1094,13 @@
 		clock-output-names = "xtal";
 	};
 
-	wifi: uccp@18480000 {
+       wifi: uccp@18480000 {
 		status = "disabled";
 		compatible = "img,pistachio-uccp";
-		reg = <0x18480000 0x38000>, <0x184BC000 0x2800>, <0x1a000000 0x00066cc0>;
-		reg-names = "uccp_core_base", "uccp_slave_base" , "uccp_pkd_gram_base";
+		reg = <0x18480000 0x9000>, <0x184C0000 0x80000>,
+		      <0x1a000000 0x00066CC0>;
+		reg-names = "uccp_sysbus_base", "uccp_perip_base",
+			    "uccp_pkd_gram_base";
 		interrupts = <GIC_SHARED 67 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "uccpirq";
 		clocks = <&clk_core CLK_RPU_CORE>,
@@ -1086,7 +1115,8 @@
 			<&clk_core CLK_AUX_ADC_INTERNAL>;
 		clock-names = "rpu_core", "rpu_v", "rpu_l", "rpu_sleep",
 				"wifi_adc", "wifi_dac", "event_timer",
-				"sys_event_timer", "aux_adc", "aux_adc_internal";
+				"sys_event_timer", "aux_adc",
+				"aux_adc_internal";
 		assigned-clocks = <&clk_core CLK_RPU_L_DIV>,
 			<&clk_core CLK_RPU_L_MUX>,
 			<&clk_core CLK_RPU_L_PLL_MUX>,
-- 
1.9.1

